userspace/libsinsp/container_info.h

```diff
@@ -122,6 +125,58 @@ class sinsp_container_info
 		std::string m_propagation;
 	};
 
+	class container_health_probe
+	{
+	public:
+
+		// The type of health probe
+		enum probe_type {
+			PT_NONE = 0,
+			PT_HEALTHCHECK,
+			PT_LIVENESS_PROBE,
+			PT_READINESS_PROBE
+		};
+
+		container_health_probe();
+		container_health_probe(const container_health_probe &other);
+		container_health_probe(container_health_probe &&other);
+		virtual ~container_health_probe();
+
+		container_health_probe &operator=(const container_health_probe &other);
+
+		// Attempt to parse any health probes out of the
+		// provided config obj and add them to the provided
+		// list.
+		static void add_health_probes(const Json::Value &config_obj,
+					      std::list<container_health_probe> &probes);
+
+		void add_to_json(Json::Value &container_obj) const;
```

Comment:
`const Json::Value &container_obj`?

Reply:
The argument shouldn't be const, as the method adds properties to the argument. The method is const as nothing about the object itself is modified, though.